### PR TITLE
Add customer parsing logic

### PR DESCRIPTION
**Pull request by yikyak02**
Implement Customer class, AddCustomerCommand, and AddCustomerCommandParser

- Add the `Customer` class, extending from `Person`, with additional fields:
  - `openCustomerOrders`: to store a list of unfulfilled customer orders.
  - `information`: for storing additional customer-specific details.
  
- Implement `AddCustomerCommand` to handle adding customers to the address book.
  - Handles execution logic for creating and adding a customer.
  
- Implement `AddCustomerCommandParser` to parse CLI input for adding customers.
  - Parses fields such as name, phone, email, address, dietary preference, information, and optional tags.
  
- Update CLI logic to support the `add customer` command.
  - Command format: `add customer n/NAME p/PHONE e/EMAIL a/ADDRESS pref/PREFERENCE i/INFORMATION [t/TAG]...`
  - Example usage: `add customer n/Emily Tan p/81234567 e/emily@example.com a/456 Cupcake Road pref/No gluten i/Allergic to nuts t/loyal`
  
rendering of customer information onto the gui is not updated yet and will be done in the next iteration.

### CODE REVIEW

**Review by bbryant824 — APPROVED** [review]
I believe the current version is good enough at this stage.

File: src/main/java/seedu/address/logic/parser/AddCustomerCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_PREFERENCE, PREFIX_INFORMATION)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCustomerCommand.MESSAGE_USAGE));
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_PREFERENCE, PREFIX_INFORMATION);
+        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
```
Comment (calebchongsj) [inline src/main/java/seedu/address/logic/parser/AddCustomerCommandParser.java:36]: Great to see that you have parsed the relevant information for a new customer

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -48,6 +70,10 @@ public Command parseCommand(String userInput) throws ParseException {
         case AddCommand.COMMAND_WORD:
             return new AddCommandParser().parse(arguments);
 
+        // Add this case for the "add customer" command
+        case AddCustomerCommand.COMMAND_WORD:
```
Comment (calebchongsj) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:74]: Good to see that you have added the relevant command within the address book.

File: src/main/java/seedu/address/model/person/Information.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    public final String value;
+
+    /**
+     * Constructs an {@code Information}.
+     *
+     * @param information A valid information string.
+     */
+    public Information(String information) {
+        requireNonNull(information);
+        checkArgument(isValidInformation(information), MESSAGE_CONSTRAINTS);
```
Comment (calebchongsj) [inline src/main/java/seedu/address/model/person/Information.java:29]: It is better to streamline customer's order history and favorite items within customer's information

**Review by calebchongsj — COMMENTED** [review]
LGTM. Amazing job to add customer into BakeBuddy.

**Review by calebchongsj — COMMENTED** [review]
Well done!

**Pull request merged**